jwst/assign_wcs/nirspec.py

```diff
@@ -256,29 +256,85 @@ def get_open_fixed_slits(input_model):
+
+        ('slitlet_id', '>i2'),
+        ('msa_metadata_id', '>i2'),
+        ('shutter_quadrant', '>i2'),
+        ('shutter_row', '>i2'),
+        ('shutter_column', '>i2'),
+        ('source_id', '>i2'),
+        ('background', 'S1'),
+        ('shutter_state', 'S6'),
+        ('estimated_source_in_shutter_x', '>f4'),
+        ('estimated_source_in_shutter_y', '>f4')])
+
+    For example, something like:
+        (12, 2, 4, 251, 22, 1, 'Y', 'OPEN', nan, nan),
 
     Parameters
     ----------
-    msa_status : str
-        Name of MSA status file.
-
-    Returns
-    -------
-    msa_slits_id : ndarray
-        An array of slit ids.
-        A slit_id is a tuple of (quadrant_number, slit_number).
-
-    """
-    with fits.open(msa_status) as fmsa:
-        msa_slits_id = []
-        for i in range(1, 5):
-            quadrant = fmsa[i].data
-            open_shutters = quadrant[quadrant['status'].nonzero()]
-            msa_slits_id.extend([(i, n) for n in open_shutters['NO']])
-    return np.array(msa_slits_id)
+        msa_file: Fits file name that contains the configuration information
+        msa_metadata_id: The MSA meta id that we are going to work on...
+    """
+
+    slitlets = []
+
+    # If they passed in a string then we shall assume it is the filename
+    # of the configuration file.
+    with fits.open(msa_file) as msa_file:
+        # Get the configuration header from teh _msa.fits file.  The EXTNAME should be 'SHUTTER_INFO'
+        msa_conf = [h for h in msa_file if 'EXTNAME' in h.header and h.header['EXTNAME'] == 'SHUTTER_INFO']
+        msa_conf = msa_conf[0]
+
+        # First we are going to filter the msa_file data on the msa_metadata_id
+        # as that is all we are interested in for this function.
+        msa_data = [x for x in msa_conf.data if x['msa_metadata_id'] == msa_metadata_id]
+
+        # First thing to do is to get the unique slitlet_ids
+        slitlet_ids_unique = list(set([x['slitlet_id'] for x in msa_data]))
+
+        # Now lets look at each unique slitlet id
+        for slitlet_id in slitlet_ids_unique:
+
+            # Get the rows for the current slitlet_id
+            ss = [x for x in msa_data if x['slitlet_id'] == slitlet_id]
+
+            # X and Y center based on where background == 'Y'
+            xy_centers = [(s['shutter_row'], s['shutter_column'])
+                          for s in ss
+                          if not s['background'] == 'N']
+
+            if len(xy_centers) >= 1:
+                # TODO: Might need to reverse these if row col are reversed
+                xcen, ycen = xy_centers[0]
+            else:
+                log.warning('WARNING: Could not find shutter that is not background.')
+                return []
+
+            # y-size
+            # TODO: might have to use shutter_column instead
+            margin = 0.05
+            jmin = min([s['shutter_row'] for s in ss])
+            jmax = max([s['shutter_row'] for s in ss])
```

Comment:
These need to look at 'shutter_column'. 'shutter_row' is always the same so `jmin` and `jmax` return the size of one shutter.